src/core/qgsvectorlayer.h

```diff
@@ -995,6 +996,15 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     long featureCount( const QString &legendKey ) const;
 
+    /**
+     * List of Id of features rendered with specified legend key. Features must be first
+     * calculated by countSymbolFeatures()
+     * \returns List of Id of features rendered by symbol or an empty list of not counted or symbol not in the list.
+     *
+     * \since QGIS 3.8
+     */
+    const QgsFeatureIds featureIds( const QString &legendKey ) const;
```

Comment:
I think that's no longer needed after now everything is switched to the featurecounter approach. Can you undo *all* changes in qgsvectorlayer.

Reply:
I disagree on that point since relying on the featurecounter caused segfaults previously. This was the only way to ensure stability since the content of the featurecounter vanished at some point.

Reply:
This is also why the featureCount is store in the vectorlayer instead of using the featurecounter directly. This pattern is already implemented for a good reason.

Reply:
For a reference you can also do
``` python 
layer = iface.activeLayer()
counter=layer.countSymbolFeatures()
counter.featureCount('1')
```
On 3.4 I get RuntimeError: wrapped C/C++ object of type QgsVectorLayerFeatureCounter has been deleted

This is not due to any changes that I made, the counter is always super volatile and is not viable for storage and retrieval.

Reply:
@m-kuhn Do you still want me to fix this or are you convinced that the counter is unreliable?